### ISSUE TITLE
Model does not learn on training(very high CE)

### ISSUE THREAD

**Issue by himsR** [CSAILVision/sceneparsing#16]
Hi, I trained the model using the given code twice(once with re-scaled images of size 384 by 384(bicubic for images and nearest for annotations)) and once without scaling. I trained for around 150,000 iterations. But in both cases with the trained snapshot weights, when I do inference on the validation images, I get blank images. Also during the training the cross entropy is very high all the time(~600000)  and doesn't seem to come down at all. So did you use the same settings given in the solver_FCN ? specifically the base_lr: 1e-10? Are there any other tricks that must be used to train the model because now the predictions are completely blank and with this high CE it's obvious that the model is not learning anything. 

Note- With the pre-trained weights that you have provided I can reproduce your results and I get 71.95% pixel accuracy using the FCN model. Just the training part does not seem to work. Also I tried initialising all the layers before fc6 with the pretrained VGG-16 weights. Any pointers highly appreciated.
Thanks!

**Comment by hangzhaomit** [comment]
@himsR You do need a good pretraining model. ImageNet pretrained VGG-16 weights should be good, but note that: you need to initialize fc6 and fc7 as well, turning fc layer weights into conv layer weights. For example, fc 4096x4096 should be resized to conv 1x1x4096x4096.

**Comment by himsR** [comment]
@hangzhaomit Thanks a lot for replying! So is there any particular way that you resize? For instance the fc6 layer in your model needs to have a weight shape of (4096x512x7x7) while the corresponding fc6 layer in vgg-16 has a shape of (4096x25,088). So do mean that just doing a 25088.reshape(512,7,7) should suffice? Also how did you initialize the other three conv layers in your model ,i.e. 'score_fr', 'score_pool4'  and  'score_pool3' ? Just a plain default caffe intialization should be fine? I made another change to get it working - I initialised the deconv layers to bilinear interpolator and make it work. But without the fc6,fc7 and these remaining three layers initialisation it gives a poor accuracy on validation set- 59%. Looking forward to your response. Thanks!

**Comment by xavierpuigf** [comment]
@himsR Yes, you should reshape the fc6 and fc7 weights to load from the VGG-16 model. The weights for score_fr, score_pool4 and score_pool3 are randomly initialized and the deconv layers should be initialized with bilinear interpolation filters, as you are doing.

**Comment by himsR** [comment]
@xavierpuigf Thanks for your answer! I did all these three initialisations and they did the trick . I get around 71.1 percent pixel accuracy on the validation set ,training on full size images without any augmentations!
